ts/features/bonus/siciliaVola/screens/voucherGeneration/VoucherGeneratedScreen.tsx

```diff
@@ -11,63 +13,129 @@ import { GlobalState } from "../../../../../store/reducers/types";
-    primary: false,
-    bordered: true,
-    onPress: props.completed,
-    title: "Esci"
+    primary: true,
+    onPress: () => true,
+    title: I18n.t("global.genericSave")
   };
 
-  // TODO: dispatch props.generateVoucherRequest when the component is mounted
+  if (isNone(maybeVoucherRequest)) {
+    props.failure("Voucher request is None");
+    return null;
+  }
+  const voucherRequest = maybeVoucherRequest.value;
+  if (!isVoucherRequest(voucherRequest)) {
+    props.failure("Partial voucher request is not a complete voucher request");
+    return null;
+  }
 
-  // TODO: manage loading/error/timeout(SvGeneratedVoucherTimeoutScreen) state on generateVoucherRequest selector
-  return (
-    <BaseScreenComponent
-      goBack={true}
-      contextualHelp={emptyContextualHelp}
-      headerTitle={I18n.t("bonus.sv.headerTitle")}
-    >
-      <SafeAreaView
-        style={IOStyles.flex}
-        testID={"VoucherGeneratedScreen"}
-        ref={elementRef}
+  if (!isReady(remoteVoucherGenerated)) {
+    return (
+      <BaseScreenComponent
+        goBack={false}
+        contextualHelp={emptyContextualHelp}
+        headerTitle={I18n.t("bonus.sv.headerTitle")}
       >
-        <ScrollView style={[IOStyles.horizontalContentPadding]}>
-          <H1>{I18n.t("bonus.sv.voucherGeneration.voucherGenerated.title")}</H1>
-        </ScrollView>
-        <FooterWithButtons
-          type={"TwoButtonsInlineHalf"}
-          leftButton={backButtonProps}
-          rightButton={continueButtonProps}
+        <LoadingErrorComponent
+          isLoading={isLoading(remoteVoucherGenerated)}
+          loadingCaption={I18n.t("global.remoteStates.loading")}
+          onRetry={() => props.generateVoucherRequest(voucherRequest)}
         />
-      </SafeAreaView>
-    </BaseScreenComponent>
-  );
+      </BaseScreenComponent>
+    );
+  }
+
+  const voucherGenerated = remoteVoucherGenerated.value;
+
+  switch (voucherGenerated.kind) {
+    // TODO: manage conflict and other state when the final swagger is available
+    case "success":
+      return (
+        <BaseScreenComponent
+          goBack={false}
+          contextualHelp={emptyContextualHelp}
+          headerTitle={I18n.t("bonus.sv.headerTitle")}
+        >
+          <SafeAreaView style={IOStyles.flex} testID={"VoucherGeneratedScreen"}>
+            <ScrollView style={[IOStyles.horizontalContentPadding]}>
+              <H1>
+                {I18n.t("bonus.sv.voucherGeneration.voucherGenerated.title")}
+              </H1>
+              <View spacer />
+              <H5 weight={"Regular"}>
+                {I18n.t("bonus.sv.voucherGeneration.voucherGenerated.subtitle")}
+              </H5>
+              <View spacer />
+              <VoucherInformationComponent
+                voucherCode={voucherGenerated.value.id.toString()}
+                qrCode={voucherGenerated.value.qrCode}
+                barCode={voucherGenerated.value.barCode}
+              />
+            </ScrollView>
```

Comment:
```suggestion
              <VoucherInformationComponent
                voucherCode={voucherGenerated.value.id.toString()}
                qrCode={voucherGenerated.value.qrCode}
                barCode={voucherGenerated.value.barCode}
              />
            <View spacer large={true} />
            </ScrollView>
```
Just a minor suggestion to avoid to have the barcode too close to the footer.

Before:
<img width="379" alt="Schermata 2021-10-05 alle 17 00 51" src="https://user-images.githubusercontent.com/26501317/136049005-2bd5e4af-9ce8-4246-94d4-e1fa9faf48eb.png">

After:
<img width="393" alt="Schermata 2021-10-05 alle 17 01 19" src="https://user-images.githubusercontent.com/26501317/136049090-95602bf8-4b71-4f9c-b689-cf103c2ab1d2.png">

Reply:
Thank you, inserted spacer in [68ebf56](https://github.com/pagopa/io-app/pull/3401/commits/68ebf568214b6e21054b3c97159318fa4d10eb19)